src/Adapter/MSTest.CoreAdapter/Execution/TestMethodRunner.cs

```diff
@@ -84,6 +128,26 @@ internal UnitTestResult[] Execute()
             string inittestContextMessages = string.Empty;
 
             UnitTestResult[] result = null;
+
+            string ignoreMessage = null;
+            var isIgnoreAttributeOnClass = this.reflectHelper.IsAttributeDefined(this.testMethodInfo.Parent.ClassType, typeof(UTF.IgnoreAttribute), false);
+            var isIgnoreAttributeOnMethod = this.reflectHelper.IsAttributeDefined(this.testMethodInfo.TestMethod, typeof(UTF.IgnoreAttribute), false);
+
+            if (isIgnoreAttributeOnClass)
+            {
+                ignoreMessage = this.reflectHelper.GetIgnoreMessage(this.testMethodInfo.Parent.ClassType.GetTypeInfo());
+            }
+
+            if (string.IsNullOrEmpty(ignoreMessage) && isIgnoreAttributeOnMethod)
+            {
+                ignoreMessage = this.reflectHelper.GetIgnoreMessage(this.testMethodInfo.TestMethod);
```

Comment:
Tagging @pvlakshm to review the order - Ignore attribute on method takes precedence over class.